okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -273,16 +264,7 @@ private static Response stripBody(Response response) {
         : response;
   }
 
-  /**
-   * Called immediately before the transport transmits HTTP request headers. This is used to observe
-   * the sent time should the request be cached.
-   */
-  public void writingRequestHeaders() {
-    if (sentRequestMillis != -1) throw new IllegalStateException();
-    sentRequestMillis = System.currentTimeMillis();
-  }
-
-  boolean permitsRequestBody(Request request) {
+  static boolean permitsRequestBody(Request request) {
```

Comment:
Inline?

Looks like you might have a caller still using this as an instance method too.